services/blockstorage/sync/block_sync.go

```diff
@@ -125,26 +125,32 @@ func (bs *BlockSync) syncLoop(ctx context.Context) {
 
 func (bs *BlockSync) HandleBlockCommitted(ctx context.Context) {
 	ctx, cancel := context.WithTimeout(ctx, bs.config.BlockSyncNoCommitInterval()/2)
-	if bs.currentState != nil {
-		bs.currentState.blockCommitted(ctx)
+	defer cancel()
+
+	cs := bs.currentState // careful! we're reading a shared variable here from a different goroutine
```

Comment:
i am not sure this actually solves the issue, see: https://play.golang.org/p/1zYQyW0RjVZ

Reply:
unless I'm missing something, I think it does:
https://play.golang.org/p/--QHABNPwk9